webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/workflows/CommitAndDeployChangeRequestService.scala

```diff
@@ -111,6 +115,7 @@ class CommitAndDeployChangeRequestServiceImpl(
   , xmlSerializer         : XmlSerializer
   , xmlUnserializer       : XmlUnserializer
   , sectionSpecParser     : SectionSpecParser
+  , queryProcessor        : QueryProcessor
```

Comment:
I think you should use `DynGroupUpdaterService` which already has the logic and not the Query processor directly

And maybe we should do it at repository level in update function, to treat all call to update correctly, but maybe not since a repository should not modify data

Reply:
the DynGroupUpdaterService takes the group within ldap - so it would need to save first the group, with invalid node list (hence invalid nodelist) and then call update on it

Reply:
But you could do something like this: https://github.com/VinceMacBuche/rudder/commit/40671095ad721e93363eb8308b0f614b8e7f6109#diff-49b7910e1c5346540a45122d00fed7e2R94

Reply:
ha, indeed, it was not existent when I was coding it